agreement_legal_sale_fieldservice/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _action_confirm(self):
+        res = super(SaleOrder, self)._action_confirm()
+        for order in self:
+            agreement_id = self.env['agreement'].\
+                search([('code', '=', order.name)])
+            agreement_id.fsm_location_id = order.partner_id.\
+                service_location_id
```

Comment:
What if we have a delivery address different from the customer main address? I so, should we use it instead?

Reply:
@dreispt The client requested the fsm_location so I think we should stick with that for now, if someone raises an issue regarding this then I would be willing to look into solutions for that workflow

Reply:
@osi-scampbell  We have the Service Location on the sale order. Why not using it?

Reply:
Tests on this line result in a singleton error: https://travis-ci.org/OCA/contract/jobs/652175553#L2030